### PR TITLE
LPD-33318 Apply ERC to Announcements widget config

### PR DESCRIPTION
**Pull request by marcogalluzzi**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-33318

Move current Announcements portlet preferences using IDs to use ERCs.

# How am I fixing it?

Replacing portlet preferences in widget JSP and DisplayContext and add an upgrade process.

# How can you verify that it works?

Functional tests
`cd modules/test/playwright`
`npm run playwright -- test -g 'The distribution scopes of the Announcements widget can be configured'`

Integration tests
`cd modules/apps/announcements/announcements-test`
`gw tI --tests UpgradePortletPreferencesTest`

### CODE REVIEW

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Review by ambrinchaudhary — APPROVED** [review]
LGTM. 
My only recommendation is to use `test.step()` in the functional test where you added the comment blocks, just to make it more readable (already discussed this via slack).

**Comment by marcogalluzzi** [comment]
> LGTM. My only recommendation is to use `test.step()` in the functional test where you added the comment blocks, just to make it more readable (already discussed this via slack).

Hi @ambrinchaudhary see this commit for the added steps into Playwright: [eebab19](https://github.com/liferay-content-management/liferay-portal/pull/6094/commits/eebab194cc86d7aa59649cba4cc92701e2dcd53e)

File: modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/display/context/AnnouncementsDisplayContext.java [hunk 1]
```diff
@@ -490,34 +472,107 @@ private PortletURL _getPortletURL() {
...
+					_getSelectedScopeGroupExternalReferenceCodes())) {
+
+			groups.add(
+				GroupLocalServiceUtil.getGroupByExternalReferenceCode(
+					externalReferenceCode,
+					_announcementsRequestHelper.getCompanyId()));
+		}
+
+		return groups;
+	}
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/display/context/AnnouncementsDisplayContext.java]: Done! See new 5 commits for changes.

File: modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/display/context/AnnouncementsDisplayContext.java [hunk 2]
```diff
@@ -399,31 +377,35 @@ public boolean isCustomizeAnnouncementsDisplayed() {
...
 
 	public boolean isScopeGroupSelected(Group scopeGroup) {
-		String selectedScopeGroupIds = _getSelectedScopeGroupIds();
+		String selectedScopeGroupExternalReferenceCodes =
+			_getSelectedScopeGroupExternalReferenceCodes();
 
-		return selectedScopeGroupIds.contains(
-			String.valueOf(scopeGroup.getGroupId()));
+		return selectedScopeGroupExternalReferenceCodes.contains(
+			scopeGroup.getExternalReferenceCode());
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/display/context/AnnouncementsDisplayContext.java]: Done! See new 5 commits for changes.

Finally I'm saving the list of ERCs as a JSON array.

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
ci:forward

**Pull request closed without merging**